pkg/agent/controller/bgp/controller.go

```diff
@@ -0,0 +1,954 @@
+		serviceListerSynced:       serviceInformer.Informer().HasSynced,
+		egressInformer:            egressInformer.Informer(),
+		egressLister:              egressInformer.Lister(),
+		egressListerSynced:        egressInformer.Informer().HasSynced,
+		bgpPolicyInformer:         bgpPolicyInformer.Informer(),
+		bgpPolicyLister:           bgpPolicyInformer.Lister(),
+		bgpPolicyListerSynced:     bgpPolicyInformer.Informer().HasSynced,
+		endpointSliceInformer:     endpointSliceInformer.Informer(),
+		endpointSliceLister:       endpointSliceInformer.Lister(),
+		endpointSliceListerSynced: endpointSliceInformer.Informer().HasSynced,
+		k8sClient:                 k8sClient,
+		bgpPeerPasswords:          make(map[string]string),
+		nodeName:                  nodeConfig.Name,
+		enabledIPv4:               networkConfig.IPv4Enabled,
+		enabledIPv6:               networkConfig.IPv6Enabled,
+		podIPv4CIDR:               nodeConfig.PodIPv4CIDR.String(),
+		podIPv6CIDR:               nodeConfig.PodIPv6CIDR.String(),
+		nodeIPv4Addr:              nodeConfig.NodeIPv4Addr.IP.String(),
+		egressEnabled:             features.DefaultFeatureGate.Enabled(features.Egress),
+		newBGPServerFn: func(globalConfig *bgp.GlobalConfig) bgp.Interface {
+			return gobgp.NewGoBGPServer(globalConfig)
+		},
+		queue: workqueue.NewNamedRateLimitingQueue(workqueue.NewItemExponentialFailureRateLimiter(minRetryDelay, maxRetryDelay), "bgpPolicyGroup"),
+	}
+	c.bgpPolicyInformer.AddEventHandlerWithResyncPeriod(
+		cache.ResourceEventHandlerFuncs{
+			AddFunc:    c.addBGPPolicy,
+			UpdateFunc: c.updateBGPPolicy,
+			DeleteFunc: c.deleteBGPPolicy,
+		},
+		resyncPeriod,
+	)
+	c.serviceInformer.AddEventHandlerWithResyncPeriod(
+		cache.ResourceEventHandlerFuncs{
+			AddFunc:    c.addService,
+			UpdateFunc: c.updateService,
+			DeleteFunc: c.deleteService,
+		},
+		resyncPeriod,
+	)
+	c.endpointSliceInformer.AddEventHandlerWithResyncPeriod(
+		cache.ResourceEventHandlerFuncs{
+			AddFunc:    c.addEndpointSlice,
+			UpdateFunc: c.updateEndpointSlice,
+			DeleteFunc: nil,
+		},
+		resyncPeriod,
+	)
+	if c.egressEnabled {
+		c.egressInformer.AddEventHandlerWithResyncPeriod(
+			cache.ResourceEventHandlerFuncs{
+				AddFunc:    c.addEgress,
+				UpdateFunc: c.updateEgress,
+				DeleteFunc: c.deleteEgress,
+			},
+			resyncPeriod,
+		)
+	}
+	c.nodeInformer.AddEventHandlerWithResyncPeriod(
+		cache.ResourceEventHandlerFuncs{
+			AddFunc:    nil,
+			UpdateFunc: c.updateNode,
+			DeleteFunc: nil,
+		},
+		resyncPeriod,
+	)
+	return c, nil
+}
+
+// watchSecretChanges uses watch API directly to watch for the changes of the specific Secret storing passwords of BGP
+// peers.
+func (c *Controller) watchSecretChanges(endCh <-chan struct{}) error {
+	ns := env.GetAntreaNamespace()
+	watcher, err := c.k8sClient.CoreV1().Secrets(ns).Watch(context.TODO(), metav1.SingleObject(metav1.ObjectMeta{
+		Namespace: ns,
+		Name:      types.BGPPolicySecretName,
+	}))
```

Comment:
nit: maybe declare a secretMeta, then use it in the watch and log functions below:
```
secretMeta := &metav1.ObjectMeta{
	Name:      types.BGPPolicySecretName,
	Namespace: env.GetAntreaNamespace(),
}
watcher, err := c.k8sClient.CoreV1().Secrets(secretMeta.Namespace).Watch(context.TODO(), metav1.SingleObject(*secretMeta))
...
klog.InfoS("Starting watching Secret changes", "Secret", klog.KObj(secretMeta))
```

Reply:
Done